src/sentry/spans/consumers/process_segments/convert.py

```diff
@@ -0,0 +1,103 @@
+
+
+FIELD_TO_ATTRIBUTE = {
+    "description": "sentry.raw_description",
+    "duration_ms": "sentry.duration_ms",
+    "is_segment": "sentry.is_segment",
+    "exclusive_time_ms": "sentry.exclusive_time_ms",
+    "start_timestamp_precise": "sentry.start_timestamp_precise",
+    "end_timestamp_precise": "sentry.end_timestamp_precise",
+    "is_remote": "sentry.is_remote",
+    "parent_span_id": "sentry.parent_span_id",
+    "profile_id": "sentry.profile_id",
+    "segment_id": "sentry.segment_id",
+    "received": "sentry.received",
+    "origin": "sentry.origin",
+    "kind": "sentry.kind",
+    "hash": "sentry.hash",
+}
+
+
+def convert_span_to_item(span: Span) -> TraceItem:
+    attributes: MutableMapping[str, AnyValue] = {}  # TODO
+
+    client_sample_rate = 1.0
+    server_sample_rate = 1.0
+
+    for k, v in (span.get("data") or {}).items():
+        attributes[k] = _anyvalue(v)
+
+    for k, v in (span.get("measurements") or {}).items():
+        if k is not None and v is not None:
+            if k == "client_sample_rate":
+                client_sample_rate = v["value"]
+            elif k == "server_sample_rate":
+                server_sample_rate = v["value"]
+            else:
+                attributes[k] = AnyValue(double_value=float(v["value"]))
+
+    for k, v in (span.get("sentry_tags") or {}).items():
+        if v is not None:
+            if k == "description":
+                k = "sentry.normalized_description"
+            else:
+                k = f"sentry.{k}"
+
+            attributes[k] = AnyValue(string_value=str(v))
+
+    for k, v in (span.get("tags") or {}).items():
+        if v is not None:
+            attributes[k] = AnyValue(string_value=str(v))
+
+    for field_name, attribute_name in FIELD_TO_ATTRIBUTE.items():
+        if span.get(field_name) is not None:
+            attributes[attribute_name] = _anyvalue(span[field_name])
+
+    return TraceItem(
+        organization_id=span["organization_id"],
+        project_id=span["project_id"],
+        trace_id=span["trace_id"],
+        item_id=int(span["span_id"], 16).to_bytes(16, "little"),
+        item_type=TraceItemType.TRACE_ITEM_TYPE_SPAN,
+        timestamp=_timestamp(span["start_timestamp_precise"]),
+        attributes=attributes,
+        client_sample_rate=client_sample_rate,
+        server_sample_rate=server_sample_rate,
+        retention_days=span["retention_days"],
+        received=_timestamp(span["received"]),
+    )
+
+
+def _anyvalue(value: Any) -> AnyValue | None:
+    if value is None:
+        return None
+    elif isinstance(value, str):
+        return AnyValue(string_value=value)
+    elif isinstance(value, bool):
+        return AnyValue(bool_value=value)
+    elif isinstance(value, int):
+        return AnyValue(int_value=value)
+    elif isinstance(value, float):
+        return AnyValue(double_value=value)
+    elif isinstance(value, (list, dict)):
+        return AnyValue(string_value=orjson.dumps(value).decode())
```

Comment:
@phacops do we even support such fields and is this the right way? This follows your implementation in Relay, but I noticed in Python there are explicit constructors and there are no tests in EAP for such fields.

Reply:
We do not support those types yet but will at some point.

I think it's fine for you to still send us kvlist and array types as any value supports it and we'll encode it in our consumer. Once we're ready to support those types, we'll plan for a change.